### PR TITLE
Add test script for comparing output

### PR DESCRIPTION
**Pull request by dmargala**
Adding a script to compare the output from different versions of spex or desi_extract_spectra. It takes two fits files as input and prints out the percentage of pixels within `np.isclose` and the cumulative distribution of the pull statistic `(f_a - f_b)/sqrt(var_a + var_b)`.

```
$ python py/gpu_specter/test/parity.py -a $SCRATCH/spex_extract.fits -b $SCRATCH/desi_extract.fits
(f_a, f_b):
  isclose:   70.17%
(f_a - f_b)/sqrt(var_a + var_b):
    1e-05:   60.38%
   0.0001:   73.73%
    0.001:   84.80%
     0.01:   94.74%
      0.1:   99.88%
      1.0:  100.00%
```

I've also added some notes to the test subdirectory to codify some example testing practices.

### CODE REVIEW

File: py/gpu_specter/test/parity.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import matplotlib.pyplot as plt
+import matplotlib.colors as mcolors
+
+import fitsio
+
+def read_frame(filename):
+    """
+    Simple fits file reader
+
+    TODO: move to gpu_specter.io
```
Comment (sbailey) [inline py/gpu_specter/test/parity.py:18]: Would you like to address this TODO prior to merging?  Moving `read_frame` to `gpu_specter.io` would be useful for the spex script too.

File: py/gpu_specter/test/parity.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    diff = (np.sqrt(avar) - np.sqrt(bvar)) / np.sqrt(avar + bvar)
+    tstart, tstop = -5, 0
+    thresholds = np.logspace(tstart, tstop, (tstop - tstart) + 1, endpoint=True)
+    for threshold in thresholds:
+        fraction = np.average(np.abs(diff).ravel() < threshold)
+        print(template.format(label=threshold, fraction=fraction))
+
+
+if __name__ == "__main__":
+    main()
```
Comment (sbailey) [inline py/gpu_specter/test/parity.py:140]: For the record: our standard DESI style is to put anything that is expected to be run from the command line in bin/ (and without a .py extension), and have that wrap library code to be imported in py/ and avoid `if __name__ == "__main__":` stuff under py/.  However, that can result in polluting the $PATH namespace with a bunch of specialized one-off scripts, so I'm sympathetic to having specialized scripts like this "hidden" within py/ as long as the procedures are documented.

i.e. this is fine as is, but commenting on it stylistically as we consider where future stuff goes and how this should be organized when it is ported back to specter and desispec.

File: py/gpu_specter/test/parity.py (hunk 1 above)
Comment (dmargala) [reply py/gpu_specter/test/parity.py]: done!

File: py/gpu_specter/test/parity.py (hunk 2 above)
Comment (dmargala) [reply py/gpu_specter/test/parity.py]: Moving to bin/ at the expense of polluting $PATH seems like a reasonable tradeoff for now.

	renamed:    py/gpu_specter/test/parity.py -> bin/compare-frame
	renamed:    py/gpu_specter/test/README.md -> doc/testing.md

**Pull request merged**